### PR TITLE
A4A: Add content for migrations overview page

### PR DESCRIPTION
**Pull request by nateweller**
Resolves https://github.com/Automattic/automattic-for-agencies-dev/issues/473

## Proposed Changes

* Adds content to the `/migrations` page based on the designs: xT3ZmNjgcujghl1kQQEJzE-fi-3001_1207
* Moves `MobileSidebarNavigation` to the header actions per https://github.com/Automattic/wp-calypso/pull/90676/files#r1599251602

## Why are these changes being made?

* https://github.com/Automattic/automattic-for-agencies-dev/issues/473

## Testing Instructions

* Navigate to the `/migrations` page and compare the content against the Figma design.
* Review the page on different screen sizes.
* Validate the Tipalti status is correct for your account, and that the button navigates to the existing Tipalti invite.

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

## Screenshots

Screenshot 2024-05-13 at 2 59 38 PM

### CODE REVIEW

File: client/a8c-for-agencies/sections/migrations/migrations-overview/index.tsx [hunk 1]
```diff
@@ -1,26 +1,155 @@
...
+	const { data } = useGetTipaltiPayee();
+	const accountStatus = getAccountStatus( data, translate );
+	const statusProps = {
+		children: accountStatus?.status,
+		type: accountStatus?.statusType,
+		tooltip: accountStatus?.statusReason,
+	};
+
+	const onAddBankDetailsClick = useCallback( () => {
+		dispatch( recordTracksEvent( 'calypso_a4a_referrals_add_bank_details_button_click' ) );
```
Comment (yashwin) [inline client/a8c-for-agencies/sections/migrations/migrations-overview/index.tsx]: Can we keep the event name different for this?

```suggestion
		dispatch( recordTracksEvent( 'calypso_a4a_migrations_add_bank_details_button_click' ) );
```

File: client/a8c-for-agencies/sections/migrations/migrations-overview/index.tsx [hunk 2]
```diff
@@ -1,26 +1,155 @@
...
+	}, [ dispatch ] );
+
 	return (
-		<Layout title={ title } wide sidebarNavigation={ <MobileSidebarNavigation /> }>
+		<Layout className="referrals-layout" title={ title } wide>
```
Comment (yashwin) [inline client/a8c-for-agencies/sections/migrations/migrations-overview/index.tsx]: Can we match the component name for class names?

```suggestion
		<Layout className="migrations-overview__layout" title={ title } wide>
```

File: client/a8c-for-agencies/sections/migrations/migrations-overview/style.scss [hunk 3]
```diff
@@ -0,0 +1,108 @@
+@import "@wordpress/base-styles/breakpoints";
+@import "@wordpress/base-styles/mixins";
+
+.referrals-layout .a4a-layout__body-wrapper {
```
Comment (yashwin) [inline client/a8c-for-agencies/sections/migrations/migrations-overview/style.scss]:
```suggestion
.migrations-overview__layout .a4a-layout__body-wrapper {
```

**Review by yashwin — COMMENTED** [review]
Overall looks good. 

Some minor suggestions. Let me know what you think.

**Comment by nateweller** [comment]
@yashwin Excellent catches, thanks very much! I've committed your suggestions. :+1:

File: client/a8c-for-agencies/sections/migrations/migrations-overview/index.tsx (hunk 2 above)
Comment (jkguidaven) [reply client/a8c-for-agencies/sections/migrations/migrations-overview/index.tsx]: @nateweller I think you miss this one.

**Review by jkguidaven — APPROVED** [review]
Works great. Changes looks good to me 👍 nice work!!

**Pull request merged**